### PR TITLE
Do not emit array change events when app is being destroyed

### PR DESCRIPTION
**Pull request by igorT**
Speeds up test runs, and also makes app tests more robust against bad teardown states.

### CODE REVIEW

**Review by spham92 — APPROVED** [review]

File: addon/record-data.js [hunk 1]
```diff
@@ -581,6 +581,10 @@ export default class M3RecordData {
 
   removeFromRecordArrays() {
     if (CUSTOM_MODEL_CLASS) {
+      // No need to remove us from record arrays if the store is beeing destroyed
+      if (this.globalM3CacheRD._storeIsDestroying) {
+        return;
+      }
```
Comment (hjdivad) [inline addon/record-data.js:587]: This seems like an odd place to stash a store destroying flag. Does the store wrapper not know whether the store is being destroyed?

Comment (igorT) [reply addon/record-data.js:587]: No, there is no public API for it, so I made one we create. Any ideas for a better place to put it?

Comment (hjdivad) [reply addon/record-data.js:587]: putting the flag on the record data as a workaround seems fine to me as long as we agree that this is a deficiency in the public API that should be sorted out

Comment (igorT) [reply addon/record-data.js]: Moved to using schema's `isDestroying` flag

**Review by hjdivad — APPROVED** [review]

**Pull request merged**